### PR TITLE
Add begin() and abort() to Two-phase Commit Transaction API

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds `begin()` and `abort()` methods to Two-phase Commit Transaction API. I'll add inline comments for the details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransaction.java [hunk 1]
```diff
@@ -104,4 +105,17 @@ public interface TwoPhaseCommitTransaction extends TransactionCrudOperable {
    * @throws RollbackException if the operation fails
    */
   void rollback() throws RollbackException;
+
+  /**
+   * Aborts a transaction. This method is an alias of {@link #rollback()}.
+   *
+   * @throws AbortException if the operation fails
+   */
+  default void abort() throws AbortException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransaction.java:114]: Added `abort()` to `TwoPhaseCommitTransaction` which is an alias of `rollback()`.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 2]
```diff
@@ -61,23 +61,45 @@ public interface TwoPhaseCommitTransactionManager {
...
   /**
-   * Starts a new transaction. This method is assumed to be called by a coordinator process.
+   * Begins a new transaction. This method is assumed to be called by a coordinator process.
    *
-   * @return {@link TwoPhaseCommitTransaction}
+   * @return {@link DistributedTransaction}
    * @throws TransactionException if starting the transaction failed
    */
-  TwoPhaseCommitTransaction start() throws TransactionException;
+  TwoPhaseCommitTransaction begin() throws TransactionException;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:69]: Also added `begin()` to `TwoPhaseCommitTransactionManager`.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 3]
```diff
@@ -61,23 +61,45 @@ public interface TwoPhaseCommitTransactionManager {
...
+   * Begins a new transaction with the specified transaction ID. It is users' responsibility to
    * guarantee uniqueness of the ID so it is not recommended to use this method unless you know
    * exactly what you are doing. This method is assumed to be called by a coordinator process.
    *
    * @param txId an user-provided unique transaction ID
-   * @return {@link TwoPhaseCommitTransaction}
+   * @return {@link DistributedTransaction}
+   * @throws TransactionException if starting the transaction failed
+   */
+  TwoPhaseCommitTransaction begin(String txId) throws TransactionException;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:80]: Ditto.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 4]
```diff
@@ -61,23 +61,45 @@ public interface TwoPhaseCommitTransactionManager {
...
+
+  /**
+   * Starts a new transaction. This method is an alias of {@link #begin()}.
+   *
+   * @return {@link DistributedTransaction}
+   * @throws TransactionException if starting the transaction failed
+   */
+  default TwoPhaseCommitTransaction start() throws TransactionException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:88]: After this change, this `start()` will be regarded as an alias of `begin()`.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 5]
```diff
@@ -61,23 +61,45 @@ public interface TwoPhaseCommitTransactionManager {
...
+  /**
+   * Starts a new transaction with the specified transaction ID. This method is an alias of {@link
+   * #begin(String)}.
+   *
+   * @param txId an user-provided unique transaction ID
+   * @return {@link DistributedTransaction}
    * @throws TransactionException if starting the transaction failed
    */
-  TwoPhaseCommitTransaction start(String txId) throws TransactionException;
+  default TwoPhaseCommitTransaction start(String txId) throws TransactionException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:100]: Ditto.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 6]
```diff
@@ -116,13 +138,24 @@ public interface TwoPhaseCommitTransactionManager {
...
 
   /**
-   * Aborts a given transaction.
+   * Rolls back a given transaction.
+   *
+   * @param txId a transaction ID
+   * @return {@link TransactionState}
+   * @throws TransactionException if aborting the given transaction failed
+   */
+  TransactionState rollback(String txId) throws TransactionException;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:147]: Also added `rollback()` to `TwoPhaseCommitTransactionManager`.

File: core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java [hunk 7]
```diff
@@ -116,13 +138,24 @@ public interface TwoPhaseCommitTransactionManager {
...
+
+  /**
+   * Aborts a given transaction. This method is an alias of {@link #rollback(String)}.
    *
    * @param txId a transaction ID
    * @return {@link TransactionState}
    * @throws TransactionException if aborting the given transaction failed
    */
-  TransactionState abort(String txId) throws TransactionException;
+  default TransactionState abort(String txId) throws TransactionException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java:156]: After this change, this `abort()` will be regarded as an alias of `rollback()`.

File: rpc/src/main/proto/scalardb.proto [hunk 8]
```diff
@@ -431,11 +431,17 @@ service TwoPhaseCommitTransaction {
   }
   rpc GetState(GetTransactionStateRequest) returns (GetTransactionStateResponse) {
   }
+  rpc Rollback(RollbackRequest) returns (RollbackResponse) {
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:434]: Also added the same methods (RPCs) to proto.

**Review by Torch3333 — DISMISSED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
Sorry for the late review.
LGTM! Thank you!

**Pull request merged**